### PR TITLE
Add Scala 3.0.1 support

### PR DESCRIPTION
**Pull request by dos65**
Also, dropped support for all previous 3.0.x-RC versions.

### CODE REVIEW

File: build.sbt
Comment (tgodzik) [inline build.sbt:246]: @smarter is it ok to drop them? I can't find the date that you mentioned previously.

File: build.sbt
Comment (tgodzik) [inline build.sbt:246]: Can it be oneline? Just change it if you do any further fixes, otherwise I would leave it.

**Review by tgodzik — APPROVED** [review]
Thanks for remembering about it! As soon as it is green and @smarter has approved it we should be good to merge.

**Comment by tgodzik** [comment]
`tests.ScalaToplevelLibrarySuite.scala3-compiler_3-3.0.1-sources.jar ` failure seems legit :/

File: build.sbt
Comment (smarter) [reply build.sbt:246]: Could you wait until next week to drop RC1? We had to move the final exam for our class to tomorrow (Saturday) and are still using 3.0.0-RC1 there.

Comment (smarter) [reply build.sbt:246]: In fact if you can wait until next week for the release in general this ensures that we won't run into any new exciting issue tomorrow :sweat_smile:

Comment (dos65) [reply build.sbt:246]: @smarter Sure. It's a preparation step for release. We aren't going to do release until Monday.

**Comment by dos65** [comment]
@tgodzik yep. I'm looking at it - it's a parser issue

File: build.sbt
Comment (Swoorup) [reply build.sbt:246]: can we release on nightly for bleeding edge users 😄

**Review by tgodzik — APPROVED** [review]
LGTM!

File: build.sbt
Comment (dos65) [reply build.sbt:246]: @swoorup Unfortunately, that's won't work as we can't publish artifacts per every nightly version.

Comment (tgodzik) [reply build.sbt:246]: it should be possible to setup an infrastructure for it, but that would require building mtags on the dotty side

**Pull request merged**
